### ISSUE TITLE
Conflict with the Splashscreen plugin when starting the app offline

### ISSUE THREAD

**Issue by pauloya** [manifoldjs/ManifoldCordova#92]
Hi,
I'm testing an iOS app, using the cordova-plugin-splashscreen to control when the splashscreen goes away programmatically. The problem is that when the app starts offline I can't hide the asplashscreen, even if I set the configuration to have "AutoHideSplashScreen" to "true", or if I call the hide method:   `navigator.splashscreen.hide();`

I've tried this both using the default offline page and a custom one.

When I reconnect the device to wifi the splashscreen hides and the remote site is shown correctly.

**Comment by boyofgreen** [comment]
Hi @pauloya  it sounds like we are not firing an event that the cordova-plugin-splashscreen is looking for.  So for clarity, the splash screen shows instead of the offline page, not on top of it correct?  And it fails if it is set to auto and if it is set to timed delay?
  Also, does it fix it if you add this js to the offline page:

`window.setTimeout(function () {
    navigator.splashscreen.hide();
}, 0);`

**Comment by pauloya** [comment]
I'm attaching macOS Safari debugger to the iOS device, and I see that the offline page is there.
My custom offline page loads the cordova.js
`<script type="text/javascript" src="cordova.js"></script>`

So when I try to call from the console:
`navigator.splashscreen.hide();`
nothing happens, I tried setting a breakpoint in the splashscreen ObjectiveC code and that doesn't seem to get hit when I call hide()... 
I think the issue may be related with the fact that the offline page is not correctly bound with the cordova APIs...

**Comment by pauloya** [comment]
Just to clarify, is there anything different about the offline page that prevents cordova APIs to work correctly? I noticed that when hooking the Safari devtools with the app there are 2 different instances of browser that show up, is this because the offline page is rendered inside a non standard cordova webview?

**Comment by boyofgreen** [comment]
The reason why we use two different webviews is so that when you go back online, we can just switch webviews and keep you in context.  With iOS we would often get a page refresh where we would loose the session info.
   Your assumption is correct, we don't have the cordova environment on the offline page and we should for this scenario.   If you are using the local plugin structure, you should be able to just solve it by hard coding a link to local js files.

**Comment by pauloya** [comment]
I'm sure you know this better than me, but a quick look at the code indicates the offline view is a [UIWebView] which, perhaps doesn't bind access with Cordova even though I include it. [CDVViewController] should work instead.
Also, I think you would need to inject the plugins JS in the offline page for this to work.

**Comment by Yuripetusko** [comment]
Also seeing it. If network status changes after hosted url has been loaded then offline page is shown, but If I start the app while offline then i'm stuck on splashscreen

**Comment by boyofgreen** [comment]
Hi @pauloya I do remember some issue with offline, splashscreen and cordova APIs.  @f2bo , @DaveVoyles  or @jasonshortphd     did any of you help fix a problem like this?

**Comment by Yuripetusko** [comment]
Looks like since url in `<content src="..." />` never loads and we are not loading local files where we can do `navigator.splashscreen.hide()` there's no event that hides splashscreen,

**Comment by f2bo** [comment]
> Hi @pauloya I do remember some issue with offline, splashscreen and cordova APIs. @f2bo , @DaveVoyles or @jasonshortphd did any of you help fix a problem like this?

@boyofgreen Yes. We did fix an issue (https://github.com/manifoldjs/ManifoldCordova/issues/59) with offline in iOS, but it was a different scenario. 

@pauloya Just to confirm, do you mean you see the splashscreen instead of the offline page while the device is offline?

**Comment by pauloya** [comment]
We found a workaround for this.

> @pauloya Just to confirm, do you mean you see the splashscreen instead of the offline page while the device is offline?

Yes, the problem was that the splashscreen stayed on when the device was offline.

**The workaround is to open a local HTML that does a document.location.href = "..."; to redirect to the remote host.**

We looked at the splashscreen code and it relies on a pageload event to start any logic of hiding the splashscreen even when using a delay splashscreen. If you use the splashscreen plugin and there is no page load the splashscreen will never go away.

We set up a delay of 10s on the splashscreen and we call `navigator.splashcreen.hide()` from the hosted web app. It's not the ideal solution because of the delay of 10s when the device is offline, but we can live with it.
